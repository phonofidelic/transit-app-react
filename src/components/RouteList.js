import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';
import RouteListItem from './routeListItem';

class RouteList extends Component {

	constructor(props) {
		super(props);
		this.props.fetchNearbyRoutes();
	}

	renderList() {
		return (
			<ul>
				{this.props.routes.map((route, i) => {
					return <RouteListItem key={i} name={route.name} />
				})}
			</ul>			
		);
	}

	render() {
		const { isFetching, routes } = this.props;
		return (
			<div className="route-list-container">
				route list
				{this.props.isFetching && routes.length === 0 && <div>loading...</div>}
				{routes.length > 0 && this.renderList()}
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		routes: state.routeListReducer.routes,
		isFetching: state.routeListReducer.isFetching
	}
}

export default connect(mapStateToProps, actions)(RouteList);