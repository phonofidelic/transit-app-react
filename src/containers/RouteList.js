import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';
import RouteListItem from '../components/RouteListItem';
// import RoutesData from './RoutesData';

export class RouteList extends Component {

	constructor(props) {
		super(props);

		this.props.init();
	}

	renderList(error) {
		return (
			<ul>
				{this.props.routes.map((route, i) => {
					return <RouteListItem key={route.onestop_id}
																index={i}
																onestopId={route.onestop_id}
																name={route.name}
																longName={route.tags.route_long_name}
																color={route.color}
																stops={route.stops_served_by_route.slice(0, 9)}
																selected={route.selected}
																route={route}
																className="route-list-item" />
				})}
			</ul>
		);
	}

	renderError() {
		return (
			<div>Error: {this.props.error}</div>
		);
	}

	render() {
		const { isFetching, error, routes, showTripPlanner } = this.props;
		return (
			<div className="route-list">
				{isFetching && routes.length === 0 && <div>loading...</div>}
				{routes && !showTripPlanner ? this.renderList() : null}
				{error && this.renderError(error.message)}
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		routes: state.routeListReducer.routes,
		isFetching: state.routeListReducer.isFetching,
		lastUpdated: state.routeListReducer.lastUpdated,
		error: state.routeListReducer.error,
		showTripDisplay: state.tripPlannerReducer.showTripDisplay,
		showTripPlanner: state.tripPlannerReducer.showTripPlanner
	}
}

export default connect(mapStateToProps, actions)(RouteList);


