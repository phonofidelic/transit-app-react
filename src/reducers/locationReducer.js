import { GET_USER_POS, SET_DESTINATION } from '../actiontypes';

const INITIAL_STATE = {
	userPos: null,
	destination: null
};

const locationReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case GET_USER_POS:
			return {
				...state,
				userPos: action.payload,
				uerBbox: {
					ne: {
						lat: action.payload.lat,
						lng: action.payload.lng
					},
					sw: {
						lat: action.payload.lat,
						lng: action.payload.lng
					}
				} 
			};

		case SET_DESTINATION:
			return {
				...state,
				destination: action.payload
			}

		default:
			return state;
	}
}

export default locationReducer;