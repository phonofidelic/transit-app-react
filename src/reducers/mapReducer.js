import { 
	INIT_MAP, 
	MAP_LOADED, 
	UPDATE_MAP_VIEW, 
	SET_DEST_MARKER,
	SET_TRIP_LINE } from '../actiontypes';

const INITIAL_STATE = {
	map: null, 
	mapCenter: null, 
	mapLoaded: false,
	destMarker: null,
	tripLine: null
};

const mapReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {

		case INIT_MAP: 
			return {
				...state,
				map: action.payload
			};

		case MAP_LOADED:
			return {
				...state,
				mapLoaded: true
			}

		case UPDATE_MAP_VIEW:
			return {
				...state,
				mapCenter: action.payload
			};

		case SET_DEST_MARKER:
			return {
				...state,
				destMarker: action.payload
			}

		case SET_TRIP_LINE:
			return {
				...state,
				tripLine: action.payload
			}

		default:
			return state;
	}
}

export default mapReducer;