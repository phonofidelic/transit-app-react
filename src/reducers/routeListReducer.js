import { REQUEST_ROUTES, RECIEVE_ROUTES } from '../actiontypes';

const INITIAL_STATE = { isFetching: false, routes: [] };

const routeListReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case REQUEST_ROUTES:
			return {
				...state,
				isFetching: true
			};

		case RECIEVE_ROUTES:
			return {
				...state,
				isFetching: false,
				routes: action.payload,
				lastUpdated: action.receivedAt
			}

		default:
			return state;
	}
}

export default routeListReducer