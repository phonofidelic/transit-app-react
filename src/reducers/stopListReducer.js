import { FETCH_STOPS } from '../actiontypes';;

export const INITIAL_STATE = {
	stops: []
};

const stopListReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case FETCH_STOPS:
			return {
				...state,
				stops: action.payload
			}

		default:
			return state
	}
}

export default stopListReducer;