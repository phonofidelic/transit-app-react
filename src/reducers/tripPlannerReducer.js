import { 
	SHOW_TRIP_PLANNER,
	HIDE_TRIP_PLANNER,
 	DESTINATION_INPUT_CHANGE ,
 	RECEIVE_AUTOCOMPLETE_RESULTS,
 	SELECT_DESTINATION,
 	RECEIVE_TRIP_DATA
 } from '../actiontypes';

const INITIAL_STATE = {
	showTripPlanner: false,
	destinationInput: null,
	autocompleteResults: [],
	selectedDestination: null,
	tripData: null
};

const tripPlannerReducer = (state = INITIAL_STATE, action) => {
	switch(action.type) {
		case SHOW_TRIP_PLANNER:
			return {
				...state,
				showTripPlanner: true
			}

		case HIDE_TRIP_PLANNER:
			return {
				...state,
				showTripPlanner: false
			}

		case DESTINATION_INPUT_CHANGE:
			return {
				...state,
				destinationInput: action.payload
			}

		case RECEIVE_AUTOCOMPLETE_RESULTS:
			return {
				...state,
				autocompleteResults: action.payload
			}

		case SELECT_DESTINATION:
			return {
				...state,
				selectedDestination: action.payload
			}

		case RECEIVE_TRIP_DATA:
			return {
				...state,
				tripData: action.payload
			}

		default: return state;
	}
}

export default tripPlannerReducer;
