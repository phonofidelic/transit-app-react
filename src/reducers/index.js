import { combineReducers } from 'redux';
import userReducer from './userReducer';
import routeListReducer from './routeListReducer';
import mapReducer from './mapReducer';
import stopListReducer from './stopListReducer';
import tripPlannerReducer from './tripPlannerReducer';

const rootReducer = combineReducers({
	userReducer,
	routeListReducer,
	mapReducer,
	stopListReducer,
	tripPlannerReducer
});

export default rootReducer;