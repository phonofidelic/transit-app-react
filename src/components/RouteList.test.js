import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { RouteList } from './RouteList';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {INITIAL_STATE as featureComponent } from '../reducers/routelistReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

function setup() {
	const props = {
		fetchNearbyRoutes: jest.fn(),
		routes: []
	}

	const enzymeWrapper = shallow(<RouteList {...props} />);

	return {
		props,
		enzymeWrapper
	}
}

describe('<RouteList />', () => {
	it('renders with default props', () => {

		const { enzymeWrapper } = setup();

		expect(enzymeWrapper).toMatchSnapshot();
	});
});