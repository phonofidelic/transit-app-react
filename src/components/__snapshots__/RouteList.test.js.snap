exports[`<RouteList /> renders with default props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="route-list-container">
    route list
</div>,
  "nodes": Array [
    <div
      className="route-list-container">
      route list
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RouteList
        fetchNearbyRoutes={[Function]}
        routes={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RouteList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "fetchNearbyRoutes": [Function],
          "routes": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="route-list-container">
          route list
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="route-list-container">
          route list
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RouteList
    fetchNearbyRoutes={[Function]}
    routes={Array []} />,
}
`;
